### PR TITLE
Fixes error handling code. Fixes bug of Authenticator not launching

### PR DESCRIPTION
**Pull request by asgarli**

### CODE REVIEW

File: u2f-gae-demo/war/js/u2f-api.js [hunk 1]
```diff
@@ -225,13 +227,14 @@ u2f.WrappedChromeRuntimePort_.prototype.formatSignRequest_ =
  */
 u2f.WrappedChromeRuntimePort_.prototype.formatRegisterRequest_ =
     function(signRequests, registerRequests, timeoutSeconds, reqId) {
-  return {
+    var req = {
```
Comment (leshi) [inline u2f-gae-demo/war/js/u2f-api.js]: revert this change.

File: u2f-gae-demo/war/js/u2f-api.js [hunk 2]
```diff
@@ -185,7 +185,9 @@ u2f.getChromeRuntimePort_ = function(callback) {
  * @private
  */
 u2f.getAuthenticatorPort_ = function(callback) {
-  callback(new u2f.WrappedAuthenticatorPort_());
+  setTimeout(function() {
```
Comment (leshi) [inline u2f-gae-demo/war/js/u2f-api.js]: do you know why this change is needed?

File: u2f-gae-demo/war/js/u2f-api.js [hunk 3]
```diff
@@ -345,23 +348,23 @@ u2f.WrappedAuthenticatorPort_.prototype.doResponseFixups_ =
...
+    for (var i = 0; i < challengesList.length; i++) {
+	  var challengeObject = challengesList[i];
+	  if (challengeObject['keyHandle'] == responseObject['keyHandle']) {
+	    requestChallengeObject = challengeObject;
+	    break;
+	  }
+	}
   }
   var responseData = {
+	'errorCode': responseObject['resultCode'],
```
Comment (leshi) [inline u2f-gae-demo/war/js/u2f-api.js]: this seems wrong

Comment (leshi) [inline u2f-gae-demo/war/js/u2f-api.js]: please fix the indentation

File: u2f-gae-demo/war/js/u2f-api.js [hunk 4]
```diff
@@ -345,23 +348,23 @@ u2f.WrappedAuthenticatorPort_.prototype.doResponseFixups_ =
...
-   if (challengeObject['keyHandle'] == responseObject['keyHandle']) {
-     requestChallengeObject = challengeObject;
-     break;
-   }
- }
-  if (requestChallengeObject == null) {
-    return null;
+  /* Non-conformant sign response, do fixups. */
+  var encodedChallengeObject = responseObject['challenge'];
+  if (typeof encodedChallengeObject !== 'undefined') {
```
Comment (leshi) [inline u2f-gae-demo/war/js/u2f-api.js]: why do this instead of checking equivalence to ```null```?

File: u2f-gae-demo/war/js/u2f-api.js (hunk 2 above)
Comment (asgarli) [inline u2f-gae-demo/war/js/u2f-api.js]: No idea - I just know that Authenticator app does not get launched without this change.

File: u2f-gae-demo/war/js/u2f-api.js (hunk 4 above)
Comment (asgarli) [inline u2f-gae-demo/war/js/u2f-api.js]: I had situations where encodedChallengeObject was null when I had errors.

File: u2f-gae-demo/war/js/u2f-api.js [hunk 5]
```diff
@@ -225,7 +227,7 @@ u2f.WrappedChromeRuntimePort_.prototype.formatSignRequest_ =
  */
 u2f.WrappedChromeRuntimePort_.prototype.formatRegisterRequest_ =
     function(signRequests, registerRequests, timeoutSeconds, reqId) {
-  return {
+    return {
```
Comment (leshi) [inline u2f-gae-demo/war/js/u2f-api.js]: I think the previous indent was correct. No?

Comment (asgarli) [inline u2f-gae-demo/war/js/u2f-api.js]: This file has inconsistent indentation - I'm changing it to 2 space for this segment.
